src/components/GameProgression.js

```diff
@@ -24,6 +23,7 @@ function GameProgression({game, jidGroup, onZoneClick}) {
             j=-1;
 
         if(player.group === jidGroup || GameUtils.isCurrentUserIsOwner(game.owner)) {
+            const gameProgressionForPlayerAndZone = gameProgression[player.group][selectedPlayer][selectedZone]
```

Comment:
This will cause access a -1 index in the array! move it down or check the -1 if it is necessary